### ISSUE TITLE
Collision model

### ISSUE THREAD

**Issue by tpatten** [squirrel-project/squirrel_common#254]
The collision model does not correctly match the robot model. The corners of the base go beyond the cylinder of the collision model. @philkark this probably explains why we see collisions with objects because most of the time it is precisely the corners of the base that collide. @ipa-nhg can we use a prism instead of a cylinder?

![collision](https://user-images.githubusercontent.com/9046559/37301038-cbbfe886-2627-11e8-8242-a5f2f70a7cca.png)
![visual](https://user-images.githubusercontent.com/9046559/37301050-d3935fca-2627-11e8-8acc-697be363ea72.png)
![collision2](https://user-images.githubusercontent.com/9046559/37301066-db92e100-2627-11e8-9f55-25bc698d9295.png)
![visual2](https://user-images.githubusercontent.com/9046559/37301073-e2b4c854-2627-11e8-8e15-4a3ff78c886f.png)

**Comment by ipa-nhg** [comment]
@tpatten which part do you mean? this one: https://github.com/squirrel-project/squirrel_common/pull/243/files ? the bottom of the base?

**Comment by mzillich** [comment]
Yes, the bottom of the base. Its collision model is a cylinder, but actually it has these corners, beyond a cylinder. And these are not modelled.

**Comment by ipa-nhg** [comment]
what about use the real stl model of the base?

**Comment by mzillich** [comment]
yes we could do that. we are doing it for the robotino base itself, right?
if it does not cost much runtime

**Comment by ipa-nhg** [comment]
for the shell I only got this 3d model https://github.com/squirrel-project/squirrel_common/blob/indigo_dev/robotino_description/meshes/squirrel_assembly.stl 

that we can do is also remove the robotino base stl, to reduce the RAM consumption

**Comment by mzillich** [comment]
that's a complex model. i would expect that ti impact on perforance
can we use a rectangle/blox instad od the cylinder to model the corners?

**Comment by ipa-nhg** [comment]
I cut the shell 3d model model https://github.com/ipa-nhg/squirrel_common/blob/ac8f13048d211169b7e98be3654e6985c5024f4a/robotino_description/meshes/base_shell_front_collision.stl 

what do you think? 

Here the Pr https://github.com/squirrel-project/squirrel_common/pull/256

I can draw also a rectangle, but I am afraid this will not work, it is the third model I do for the collision of the shell base.. If you need a perfect balance between a big enough collision but as small as possible I think the only solution is use the real 3d model...

But if you prefer a rectangle, please let me know the measures.

**Comment by mzillich** [comment]
@philkark is a collision model like the above ok for the planner? not too complex, run-time expenisve?

**Comment by mzillich** [comment]
anyway, yes lets try that

**Comment by tpatten** [comment]
After discussion with @philkark we think the complex mesh will be too complex and make collisions checks very slow. Can we just replace the half cylinder with a box instead?

**Comment by ipa-nhg** [comment]
yes, of course!

can you do that? or give me the measures of the box?

**Comment by tpatten** [comment]
The measure for the box is 48cm wide (we can just make it 50cm for safety). The depth and height can stay the same as the half cylinder in https://github.com/squirrel-project/squirrel_common/pull/243/files.

**Comment by ipa-nhg** [comment]
is this right??
![screenshot from 2018-03-14 15 57 55](https://user-images.githubusercontent.com/650568/37410287-94fd2c94-27a0-11e8-954f-ea75cefbc4de.png)


if yes, 50cm is not enough

**Comment by tpatten** [comment]
A little bit longer so that it covers the front part of the robot model and then this would be perfect

**Comment by ipa-nhg** [comment]
![screenshot from 2018-03-14 16 12 08](https://user-images.githubusercontent.com/650568/37411234-a1e1b3e2-27a2-11e8-8339-b96aca54a5a8.png)
![screenshot from 2018-03-14 16 12 49](https://user-images.githubusercontent.com/650568/37411241-a5fa7e46-27a2-11e8-9023-767d9e4d48c1.png)

better?

**Comment by tpatten** [comment]
Yes looks good. We will try it.

**Comment by ipa-nhg** [comment]
committed: https://github.com/squirrel-project/squirrel_common/pull/256

**Comment by mzillich** [comment]
@ipa-nhg @philkark More on collision models:

a) for the door opening the collision model is not tight enough, see the following images
![collision](https://user-images.githubusercontent.com/4312487/37481045-0041cad8-2881-11e8-9a65-7e2a32bc4de4.png)
![visual](https://user-images.githubusercontent.com/4312487/37481046-007ab208-2881-11e8-8a18-61fe849e2d5b.png)

b) I noticed the collision model for the robotino base (why is it red actually?) is much much more detailed than the visual model. I guess this unnecessarily slows down the planner. Can we simply use the simple mesh also for collision checking?

![collision2](https://user-images.githubusercontent.com/4312487/37481085-272ce196-2881-11e8-8f8d-b89a7832edc5.png)
![visual2](https://user-images.githubusercontent.com/4312487/37481087-2766ac14-2881-11e8-86da-6e9584e2ae39.png)

**Comment by ipa-nhg** [comment]
About the door. No problem, I can reduce the model. But remember that the bottom part of the real door is wider.

For the base, the real model is a collada file (with colour and textures, very heavy file and probably not supported by the planner). I can try to export the stl file of it, but I am not sure if this we be much better. Let's try!

It is red because I didn't define gazebo tags for those links, and by default gazebo spawn unknown object in red. I thought use different colours for the visual and collision can help to visualize and compare both models, but if this is not ok for you, let me know and I will add the tags.

**Comment by ipa-nhg** [comment]
For the door: https://github.com/squirrel-project/squirrel_common/pull/259
For the base: https://github.com/squirrel-project/squirrel_common/pull/258

Please test!

**Comment by tpatten** [comment]
Hi Nadia, I am still testing the base shell front collision model. We have the updated version now. Could we make it even slightly bigger? Because we are still colliding with objects. @philkark @mzillich what do you think?

**Comment by ipa-nhg** [comment]
@tpatten https://github.com/squirrel-project/squirrel_common/pull/260 

I just added a scale factor to the URDF and didn't modify the mesh file, @tpatten you can modify the factor easily for your tests

**Comment by tpatten** [comment]
Great! Thanks that helps.
